avm/res/automation/automation-account/main.bicep

```diff
@@ -25,6 +25,15 @@ param credentials credentialType = []
 @description('Optional. List of modules to be created in the automation account.')
 param modules array = []
 
+@description('Optional. List of powershell72 modules to be created in the automation account.')
+param powershell72Modules array = []
```

Comment:
Can we start using user defined types to improve the consumer experience? This module would have many applicable. For this PR I'd only suggest to add them for the new parameters

Reply:
I've updated the code with user defined types for the new parameters

Reply:
python 3, phyton 2 packages child modules are very similar, except for the call to the resource: resource python3package 'Microsoft.Automation/automationAccounts/python3Packages@2023-11-01' or resource python3package 'Microsoft.Automation/automationAccounts/python2Packages@2023-11-01'

Should we create a single child calling the right resource based on a input parameter? Or would it not be standard?

Reply:
I'm setting this PR in draft mode as i found i can use the discriminator for the user defined type